vaadin-text-field-flow-demo/src/main/java/com/vaadin/flow/component/textfield/demo/TextFieldView.java

```diff
@@ -173,6 +179,44 @@ private void textFieldFocusShortcut() {
         this.addCard("Text field", "Focus shortcut usage", textField);
     }
 
+    private void textFieldHelperText() {
+        Div div = new Div();
+        // begin-source-example
+        // source-example-heading: Helper Text and Component
+        TextField helperFieldText = new TextField("First name");
+        helperFieldText.setHelperText("Enter all your first names");
+
+        TextField helperFieldComponent = new TextField("Last Name");
+        helperFieldComponent.setHelperComponent(new Span("Family name"));
+
+        add(helperFieldText, helperFieldComponent);
+        // end-source-example
+
+        div.add(helperFieldText, new Text("  "), helperFieldComponent);
+        addCard("Text field", "Helper Text and Component", div);
+    }
+
+    private void textFieldHelperTextVariants() {
+        Div div = new Div();
+        // begin-source-example
+        // source-example-heading: Text field Helper Variant
+        TextField helperFieldBelow = new TextField();
+        helperFieldBelow.setHelperText("Helper Text displayed below the field");
+
+        TextField helperFieldAbove = new TextField();
+        helperFieldAbove.setHelperText("Helper Text displayed above the field");
+        helperFieldAbove
+              .addThemeVariants(TextFieldVariant.LUMO_HELPER_ABOVE_FIELD);
+
+        add(helperFieldBelow, helperFieldAbove);
+        // end-source-example
+
+        div.getStyle().set("display","flex");
+        helperFieldBelow.getStyle().set("margin-right","20px");
+        div.add(helperFieldBelow, helperFieldAbove);
+        addCard("Text field", "Text field Helper Variant", div);
```

Comment:
Checking a comment from Guille in another PR and also the one I made for radio-button-flow, it may make more sense to move this demo to the "Theme variants" tab. The text-area one for the same feature could be removed.

Reply:
Done :)

Reply:
Awesome! 😊